### PR TITLE
MG-2051 - Fix unable to reset user password

### LINKED ISSUE #2051: Bug: Unable to reset user password
### What were you trying to achieve?

I want to reset my user password but I get an error

### What are the expected results?

Password reset successfully

### What are the received results?

{
	"error": "",
	"message": "entity not found"
}

### Steps To Reproduce

_No response_

### In what environment did you encounter the issue?

Ubuntu 22.10
Docker 24.0.5
Golang 1.21.1

### Additional information you deem important

_No response_

### PR DESCRIPTION
**Pull request by VeddySama**
# What type of PR is this?
Fix Bug

## What does this do?
Fixed unable to reset user password

## Which issue(s) does this PR fix/relate to?

- Related Issue #2051
- Fixes #2051

### CODE REVIEW

**Review by dborovcanin — COMMENTED** [review]
Thanks for the contribution, the changes look good. Can you add tests for this method?

**Comment by VeddySama** [comment]
> Thanks for the contribution, the changes look good. Can you add tests for this method?

I have completed the tests, please take a look

File: users/service_test.go [hunk 1]
```diff
@@ -1451,6 +1451,70 @@ func TestRefreshToken(t *testing.T) {
...
+		resetToken string
+		secret     string
+		client     mgclients.Client
+		err        error
+	}{
+		{
+			desc:       "reset secret with valid reset token for an existing client",
+			resetToken: validToken,
+			client:     client,
+			secret:     "newSecret",
```
Comment (arvindh123) [inline users/service_test.go]: Please convert secret string values `newSecret`, `secret` to const

**Review by arvindh123 — CHANGES_REQUESTED** [review]

File: users/service_test.go (hunk 1 above)
Comment (VeddySama) [reply users/service_test.go]: Is this correct now?

Comment (arvindh123) [reply users/service_test.go]: Yes

**Comment by dborovcanin** [comment]
@ianmuchyri Please review. @VeddySama Can you, please, rebase?

**Comment by ianmuchyri** [comment]
> @ianmuchyri Please review. @VeddySama Can you, please, rebase?

@VeddySama please check lint error and sign off your commits. This is in line with the contribution guidelines.

**Comment by VeddySama** [comment]
> @ianmuchyri请查阅。@VeddySama你能重新建立基础吗？

Should I do it like I am doing now or should I re-establish a branch?

**Comment by VeddySama** [comment]
I saw that the latest PR has been modified to submit this issue. You can adopt this PR.
https://github.com/absmach/magistrala/pull/2135

**Comment by dborovcanin** [comment]
@VeddySama Thanks for the contribution and sorry for the confusion; @ianmuchyri accidentally sent a PR without checking if there is a pending one.

**Pull request closed without merging**